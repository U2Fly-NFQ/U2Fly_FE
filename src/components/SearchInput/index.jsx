import { Input } from 'antd'
import { useEffect, useRef, useState } from 'react'
import { useTranslation } from 'react-i18next'

const { Search } = Input

const SearchCustom = (props) => {
  const { style } = props
  const [text, setText] = useState('')
  const { t } = useTranslation()

  const firstUpdate = useRef(true)

  useEffect(() => {
    if (firstUpdate.current) {
      firstUpdate.current = false
      return
    }

    const timerId = setTimeout(() => {}, 500)

    return () => clearTimeout(timerId)
  }, [text])

  return (
    <Search
      placeholder={t('cta.place-holder-search')}
      enterButton={t('cta.search')}
      value={text}
      onChange={(e) => setText(e.target.value)}
      style={{ ...style }}
    />
  )
}

export default SearchCustom
