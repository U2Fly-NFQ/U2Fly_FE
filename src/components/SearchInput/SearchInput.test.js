import renderer from 'react-test-renderer'
import React, { Suspense } from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import SearchInput from './'
describe('Booking Passenger bar test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Suspense>
            <SearchInput style={{}} />
          </Suspense>
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
