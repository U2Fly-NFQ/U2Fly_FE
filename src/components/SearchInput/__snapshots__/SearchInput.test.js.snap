// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking Passenger bar test renders correctly 1`] = `
<span
  className="ant-input-group-wrapper ant-input-search ant-input-search-with-button"
  style={Object {}}
>
  <span
    className="ant-input-wrapper ant-input-group"
  >
    <input
      className="ant-input"
      hidden={null}
      onBlur={[Function]}
      onChange={[Function]}
      onCompositionEnd={[Function]}
      onCompositionStart={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="cta.place-holder-search"
      style={null}
      type="text"
      value=""
    />
    <span
      className="ant-input-group-addon"
    >
      <button
        className="ant-btn ant-btn-primary ant-input-search-button"
        disabled={false}
        onClick={[Function]}
        onMouseDown={[Function]}
        type="button"
      >
        <span>
          cta.search
        </span>
      </button>
    </span>
  </span>
</span>
`;
